python/lsst/pipe/tasks/postprocess.py

```diff
@@ -789,6 +791,167 @@ def _makeArgumentParser(cls):
+        multiple=True,
+    )
+    visitSummary = cT.Output(
+        doc="Consolidated visit-level exposure metadata",
+        name="visitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit"),
+    )
+
+
+class ConsolidateVisitSummaryConfig(pipeBase.PipelineTaskConfig,
+                                    pipelineConnections=ConsolidateVisitSummaryConnections):
+    """Config for ConsolidateVisitSummaryTask"""
+    pass
+
+
+class ConsolidateVisitSummaryTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
+    """Task to consolidate per-detector visit metadata
+    """
+    _DefaultName = "consolidateVisitSummary"
+    ConfigClass = ConsolidateVisitSummaryConfig
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = ArgumentParser(name=cls._DefaultName)
+
+        parser.add_id_argument("--id", "calexp",
+                               help="data ID, e.g. --id visit=12345",
+                               ContainerClass=VisitDataIdContainer)
+        return parser
+
+    def writeMetadata(self, dataRef):
+        """No metadata to write.
+        """
+        pass
+
+    def writeConfig(self, butler, clobber=False, doBackup=True):
+        """No config to write.
+        """
+        pass
+
+    def runDataRef(self, dataRefList):
+        self.isGen3 = False
+
+        visit = dataRefList[0].dataId['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefList), visit))
+
+        expCatalog = self.run(visit, dataRefList)
+
+        dataRefList[0].put(expCatalog, 'visitSummary', visit=visit)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        self.isGen3 = True
+
+        dataRefs = butlerQC.get(inputRefs.calexp)
+        visit = dataRefs[0].dataId.byName()['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefs), visit))
+
+        expCatalog = self.run(visit, dataRefs)
+
+        butlerQC.put(expCatalog, outputRefs.visitSummary)
+
+    def run(self, visit, dataRefs):
+        """Make a combined exposure catalog from a list of dataRefs.
+
+        Parameters
+        ----------
+        visit : `int`
+            Visit identification number
+        dataRefs : `list`
+            List of calexp dataRefs in visit.  May be list of
+            `lsst.daf.persistence.ButlerDataRef` (Gen2) or
+            `lsst.daf.butler.DeferredDatasetHandle` (Gen3).
+
+        Returns
+        -------
+        visitSummary : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with per-detector summary information.
+        """
+        schema = afwTable.ExposureTable.makeMinimalSchema()
+        schema.addField('visit', type='I', doc='Visit number')
+        schema.addField('ccd', type='I', doc='Detector number')
+        schema.addField('filterName', type='String', size=20, doc='Filter name')
```

Comment:
I could call it `filterLabel` in preparation for the upcoming RFC.  But I don't actually know what the final form of RFC-730 will be, so I can't future-proof against changes there.  But `filterLabel` is as good a guess as any.

Reply:
`filterLabel` is the name of the `Filter` replacement, which will contain both a `band` and a `physical_filter`, and we should just use that object. Lets get rid of "filterName" here and the "what kind of filter" ambiguity it has, and use `filterLabel` below.

Reply:
Oh, wait, it contains both `band` and `physical_filter`?  I didn't realize that.  I'm not sure how to persist that object, which doesn't exist yet anyway.  Again, I don't want to be tied to an as-yet-unimplemented RFC, so I'm not sure what can be done here.

Reply:
My point is, don't persist `filterName` at all, just persist the `Filter` object now, which will be replaced with a `FilterLabel` object once RFC-730 is implemented. Then there is no ambiguity.

Oh, shoot... looking at it again, `ExposureRecord` does not in fact contain `Filter`. Ugh. Hmm... I'm not sure what to do here without adding `Filter` to `ExposureRecord`, which is a pretty simple change, but probably not one you want to do as part of this.

Reply:
The only thing you can put in here is a string for the filter isn't it? I don't think `Filter` can be serialized at all (at least `Filter.writeFits()` does not work -- for gen3 `butler.put` I had to serialize it myself to a yaml string.

Reply:
Looping in @kfindeisen who is going to do the RFC-730 transition.  A possibility right now is to define `band` and `physical_filter`, and only fill the `physical_filter` with the current filter name right now.  Would this be consistent with what you have in mind for RFC-730?  (Not to tie you to this, but have at least some small chance of future-proofing these column names).

Reply:
You could include both names, but I wonder if you need to. AFAIK, most catalogs currently don't have any filter information, with tasks instead getting it from the data ID, exposure, etc. Could you clarify what the filter/band information in the catalog would be used for?

20 characters isn't enough if you're going with physical filter; see Slack for discussion.

Reply:
Result of [Slack discussion](https://lsstc.slack.com/archives/C2JPT1KB7/p1603403596119300) is that physical filters are capped at 32 chars (Middleware will crash otherwise), and band should match in case somebody tries to define a band that equals a long physical filter.

Reply:
I'll do 32 here.  And what will this be used for?  Convenience, and also looking forward to https://github.com/lsst/sdm_schemas/blob/master/yml/baselineSchema.yaml#L5233 .  Having this right in the table makes it easy to consolidate.

Reply:
I assume the SDM's "filter" is unambiguously physical filter? (Not my problem, but somebody needs to check.)

Reply:
SDM was written prior to RFC-730 and will need to be updated (and not just this).  I imagine we want both `physical_filter` and `band` there.